docs/authentication.md

```diff
@@ -0,0 +1,318 @@
+
+Let's look at an example for a profile page. This will initially render a loading skeleton. Once the request for a user has finished, it will show the user's name.
+
+```jsx
+// pages/profile.js
+
+export default Profile = () => {
+  // Fetch the user client-side
+  const { user } = useUser({ redirectTo: '/login' });
+
+  // Server-render loading skeleton
+  if (!user) {
+    return <LoadingSkeleton />;
+  }
+
+  // Once the user request finishes, show the user's name
+  return <p>{user.name}</p>;
+};
+```
+
+### Server-side Rendering
+
+If you export an `async` function called `getServerSideProps` from a page, Next.js will pre-render this page on each request using the data returned by `getServerSideProps`.
+
+```jsx
+export async function getServerSideProps(context) {
+  return {
+    props: {} // Will be passed to the page component as props
+  };
+}
+```
+
+Let's transform the profile example from earlier to use server-side rendering. If there's a session, return `session` as a prop to the `Profile` component in the page. Otherwise, redirect to `/login` on the server-side. Notice there is not a loading skeleton in this example.
+
+```jsx
+// pages/profile.js
+
+export default Profile = ({ session }) => {
+  return <p>{session.user.name}</p>;
+};
+
+export async function getServerSideProps({ req, res }) {
+  // Get the user's session based on the request
+  const session = await getSession(req);
+
+  if (!session) {
+    // If no user, redirect to login
+    res.writeHead(307, { Location: '/login' });
+    res.end();
+    return { props: {} };
+  }
+
+  // If there is a user, return the current session
+  return { props: { session } };
+}
+```
+
+An advantage of this pattern is preventing a flash of unauthenticated content before redirecting. It's important to note fetching user data in `getServerSideProps` will block rendering until the request to your authentication provider resolves. To prevent creating a bottleneck and decreasing your TTFB ([Time to First Byte](https://web.dev/time-to-first-byte/)), you should ensure your authentication lookup is fast. Otherwise, consider static generation.
+
+There is an [open RFC](https://github.com/vercel/next.js/discussions/14890) to improve redirecting inside `getServerSideProps`.
+
+## Providers
+
+Now that we've discussed authentication patterns, let's look at specific providers and explore how they're used with Next.js.
+
+### Firebase
+
+When using [Firebase Authentication](/blog/nextjs-firebase-serverless), we recommend using the static generation pattern. There currently isn't a way to log-in server-side using the Firebase Admin SDK.
```

Comment:
There will be a PR soon by a member of the community to update the firebase auth example to support SSR too (https://github.com/vercel/next.js/pull/15202 - by @kmjennison)

Reply:
Also `/blog/nextjs-firebase-serverless` looks unintended

Reply:
I added `with-firebase-authentication-ssr` here, assuming that PR would get merged first. I can change this if this is ready before then. Also fixed the link, good catch 👍